test/dataset/test_multiprocessing_loader.py

```diff
@@ -191,6 +202,81 @@ def test_validation_loader_equivalence() -> None:
     ), "Batches in incorrect context"
 
 
+@pytest.mark.parametrize(
+    "num_workers",
+    [i for i in [None, 1, 2, 3] if i is None or i <= mp.cpu_count()],
+)
+def test_train_loader_goes_over_all_data(num_workers) -> None:
+    batch_size = 4
+    num_batches_per_epoch = 4
+
+    X = 3
+
+    simple_data = [
+        {
+            "start": "2012-01-01",
+            "target": np.random.uniform(size=40).astype(float).tolist(),
+            "item_id": i,
+        }
+        for i in range(batch_size * num_batches_per_epoch * X)
+    ]
+
+    if num_workers:
+        assert len(simple_data) % num_workers == 0
```

Comment:
Why is this required? I guess it should work in any case, right?

Reply:
removed it